dbt-snowflake/tests/functional/extra_config/test_extra_config.py

```diff
@@ -0,0 +1,268 @@
+
+# Dynamic table model templates
+_MODELS_DYNAMIC_TABLE_ROW_ACCESS_POLICY = (
+    """
+{{ config(
+    target_lag = '7 days',
+    snowflake_warehouse = '"""
+    + os.getenv("SNOWFLAKE_TEST_WAREHOUSE", "")
+    + """',
+    materialized = 'dynamic_table',
+    row_access_policy = 'row_access_policy_name on (id)',
+) }}
+
+select 1 as id
+"""
+)
+_DDL_MODELS_DYNAMIC_TABLE_ROW_ACCESS_POLICY = (
+    r"\.dynamic_table_row_access_policy.*with row access policy row_access_policy_name on \(id\)"
+)
+
+_MODELS_DYNAMIC_TABLE_TAG = (
+    """
+{{ config(
+    target_lag = '7 days',
+    snowflake_warehouse = '"""
+    + os.getenv("SNOWFLAKE_TEST_WAREHOUSE", "")
+    + """',
+    materialized = 'dynamic_table',
+    table_tag = "tag_name = 'tag_value'",
+) }}
+
+select 1 as id
+"""
+)
+_DDL_MODELS_DYNAMIC_TABLE_TAG = r"\.dynamic_table_tag.*with tag \(tag_name = 'tag_value'\)"
+
+_MODELS_DYNAMIC_TABLE_TAG_AND_ROW_ACCESS_POLICY = (
+    """
+{{ config(
+    target_lag = '7 days',
+    snowflake_warehouse = '"""
+    + os.getenv("SNOWFLAKE_TEST_WAREHOUSE", "")
+    + """',
+    materialized = 'dynamic_table',
+    table_tag = "tag_name = 'tag_value'",
+    row_access_policy = 'row_access_policy_name on (id)',
+) }}
+
+select 1 as id
+"""
+)
+_DDL_MODELS_DYNAMIC_TABLE_TAG_AND_ROW_ACCESS_POLICY = r"\.dynamic_table_tag_and_row_access_policy.*with row access policy row_access_policy_name on \(id\).*with tag \(tag_name = 'tag_value'\)"
+
+
+class TestExtraConfig:
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            # Table models
+            "table_row_access_policy.sql": _MODELS_TABLE_ROW_ACCESS_POLICY,
+            "table_tag.sql": _MODELS_TABLE_TAG,
+            "table_tag_and_row_access_policy.sql": _MODELS_TABLE_TAG_AND_ROW_ACCESS_POLICY,
+            # View models
+            "view_row_access_policy.sql": _MODELS_VIEW_ROW_ACCESS_POLICY,
+            "view_tag.sql": _MODELS_VIEW_TAG,
+            "view_tag_and_row_access_policy.sql": _MODELS_VIEW_TAG_AND_ROW_ACCESS_POLICY,
+            # Incremental models
+            "incremental_row_access_policy.sql": _MODELS_INCREMENTAL_ROW_ACCESS_POLICY,
+            "incremental_tag.sql": _MODELS_INCREMENTAL_TAG,
+            "incremental_tag_and_row_access_policy.sql": _MODELS_INCREMENTAL_TAG_AND_ROW_ACCESS_POLICY,
+            # Dynamic table models
+            "dynamic_table_row_access_policy.sql": _MODELS_DYNAMIC_TABLE_ROW_ACCESS_POLICY,
+            "dynamic_table_tag.sql": _MODELS_DYNAMIC_TABLE_TAG,
+            "dynamic_table_tag_and_row_access_policy.sql": _MODELS_DYNAMIC_TABLE_TAG_AND_ROW_ACCESS_POLICY,
+        }
+
+    def test_extra_config_table(self, project):
+        # depending on the Snowflake edition tags and row access policies are supported so we check the DDL sent
+        results = run_dbt(["run", "--select", "table_*"], expect_pass=None)
```

Comment:
Is there a reason we wouldn't expect this to pass? And if so, we should really pass in `expect_pass=False`. `None` is "false-y" so it works, but we should match types.

Reply:
My personal test account didn't have Snowflake Enterprise Edition, so I put `None` especially so that it would pass if the dbt command is successful or not.

Reply:
just tried to run this test, looks like we need to actually create the row access policies